packages/iTwinUI-react/src/core/ComboBox/helpers.ts

```diff
@@ -63,6 +35,15 @@ export const ComboBoxStateContext = React.createContext<
 ComboBoxStateContext.displayName = 'ComboBoxStateContext';
 
 export const ComboBoxActionContext = React.createContext<
-  ((x: [ComboBoxAction] | [ComboBoxAction, number]) => void) | undefined
+  | ((
+      x:
+        | { type: 'multiselect'; value: number }
+        | { type: 'multi-override'; value: number[] }
+        | { type: 'open'; value: undefined }
+        | { type: 'close'; value: undefined }
+        | { type: 'select'; value: number }
+        | { type: 'focus'; value: number | undefined },
+    ) => void)
```

Comment:
It might not make sense to have these types in this file anymore if the reducer's types are defined in ComboBox.tsx. I wonder if it's possible to make a generic that gets inferred by the `value` we pass to the provider.